src/runtime/components/forms/SelectMenu.vue

```diff
@@ -364,39 +365,49 @@ export default defineComponent({
     })
 
     const selected = computed(() => {
-      if (props.multiple) {
-        if (!Array.isArray(props.modelValue) || !props.modelValue.length) {
-          return []
+      function compareValues(value1: any, value2: any) {
+        if (props.by && typeof value1 === 'object' && typeof value2 === 'object') {
+          return isEqual(value1[props.by], value2[props.by])
         }
+        return isEqual(value1, value2)
+      }
 
+      function getValue(value: any) {
         if (props.valueAttribute) {
-          return options.value.filter(option => (props.modelValue as any[]).includes(option[props.valueAttribute]))
+          return accessor(value, props.valueAttribute)
         }
-        return options.value.filter(option => (props.modelValue as any[]).includes(option))
+
+        return value
       }
 
-      if (props.valueAttribute) {
-        return options.value.find(option => option[props.valueAttribute] === props.modelValue)
+      if (props.multiple) {
+        const modelValue = props.modelValue
+        if (!Array.isArray(modelValue) || !modelValue.length) {
+          return []
+        }
+
+        return options.value.filter((option) => {
+          const optionValue = getValue(option)
+          return modelValue.some(value => compareValues(value, optionValue))
+        })
       }
-      return options.value.find(option => option === props.modelValue)
+
+      return options.value.find((option) => {
+        const optionValue = getValue(option)
+        return compareValues(optionValue, toRaw(props.modelValue))
+      })
     })
 
     const label = computed(() => {
-      if (props.multiple) {
-        if (Array.isArray(props.modelValue) && props.modelValue.length) {
-          return `${selected.value.length} selected`
-        } else {
-          return null
-        }
-      } else if (props.modelValue !== undefined && props.modelValue !== null) {
-        if (props.valueAttribute) {
-          return accessor(selected.value, props.optionAttribute) ?? null
-        } else {
-          return ['string', 'number'].includes(typeof props.modelValue) ? props.modelValue : accessor(props.modelValue as Record<string, any>, props.optionAttribute)
-        }
+      if (!props.modelValue) return null
+
+      if (Array.isArray(props.modelValue) && props.modelValue.length) {
+        return `${props.modelValue.length} selected`
+      } else if (['string', 'number'].includes(typeof props.modelValue)) {
```

Comment:
i think i already handle it if the `modelValue` is a string or number here @benjamincanac

Reply:
No you're displaying the `modelValue` which is wrong, take a look at https://ui-8paziklnq-nuxt-js.vercel.app/components/select-menu#objects. The second example displays `1`.